control-plane-operator/controllers/hostedcontrolplane/testdata/hosted-cluster-config-operator/zz_fixture_TestControlPlaneComponents.yaml

```diff
@@ -0,0 +1,187 @@
+            port: 6060
+            scheme: HTTP
+          initialDelaySeconds: 60
+          periodSeconds: 60
+          successThreshold: 1
+          timeoutSeconds: 5
+        name: hosted-cluster-config-operator
+        ports:
+        - containerPort: 8080
+          name: metrics
+          protocol: TCP
+        readinessProbe:
+          failureThreshold: 3
+          httpGet:
+            path: /readyz
+            port: 6060
+            scheme: HTTP
+          periodSeconds: 10
+          successThreshold: 1
+          timeoutSeconds: 5
+        resources:
+          requests:
+            cpu: 60m
+            memory: 80Mi
+        volumeMounts:
+        - mountPath: /etc/kubernetes/cluster-signer-ca
+          name: cluster-signer-ca
+        - mountPath: /etc/kubernetes/kubeconfig
+          name: kubeconfig
+        - mountPath: /etc/kubernetes/root-ca
+          name: root-ca
+      initContainers:
+      - command:
+        - /usr/bin/control-plane-operator
+        - availability-prober
+        - --target
+        - https://kube-apiserver:6443/readyz
+        - --kubeconfig=/var/kubeconfig/kubeconfig
+        - --required-api=imageregistry.operator.openshift.io,v1,Config
+        - --required-api=config.openshift.io,v1,Infrastructure
+        - --required-api=config.openshift.io,v1,DNS
+        - --required-api=config.openshift.io,v1,Ingress
+        - --required-api=config.openshift.io,v1,Network
+        - --required-api=config.openshift.io,v1,Proxy
+        - --required-api=config.openshift.io,v1,Build
+        - --required-api=config.openshift.io,v1,Image
+        - --required-api=config.openshift.io,v1,Project
+        - --required-api=config.openshift.io,v1,ClusterVersion
+        - --required-api=config.openshift.io,v1,FeatureGate
+        - --required-api=config.openshift.io,v1,ClusterOperator
+        - --required-api=config.openshift.io,v1,OperatorHub
+        - --required-api=operator.openshift.io,v1,Network
+        - --required-api=operator.openshift.io,v1,CloudCredential
+        - --required-api=operator.openshift.io,v1,IngressController
+        image: availability-prober
+        imagePullPolicy: IfNotPresent
+        name: availability-prober
+        resources: {}
+        volumeMounts:
+        - mountPath: /var/kubeconfig
+          name: kubeconfig
+      priorityClassName: hypershift-control-plane
+      schedulerName: default-scheduler
+      serviceAccount: hosted-cluster-config-operator
+      serviceAccountName: hosted-cluster-config-operator
+      tolerations:
+      - effect: NoSchedule
+        key: hypershift.openshift.io/control-plane
+        operator: Equal
+        value: "true"
+      - effect: NoSchedule
+        key: hypershift.openshift.io/cluster
+        operator: Equal
+        value: hcp-namespace
+      volumes:
+      - name: kubeconfig
+        secret:
+          defaultMode: 416
+          secretName: hcco-kubeconfig
+      - configMap:
+          defaultMode: 420
```

Comment:
nit: we should be consistent with defaultMode (looks like we use 416 and 420 all over the place), in this deployment we have 2 configmaps with different defaultMode

Reply:
I think this should be possibly enforced by code. Not to block this PR.

Reply:
makes sense, added a centralized function to enforce the same defaultMode.